doc/usingtesteth.rst

```diff
@@ -0,0 +1,111 @@
+
+*****************************************************
+Using Testeth
+*****************************************************
+
+Ethereum cpp-client testeth tool for creation and execution of ethereum tests.
+
+To run tests you should open folder (see also `Installing and building <https://github.com/ethereum/cpp-ethereum#building-from-source>`_)
+
+   ``/build/test``
+
+and execute a command:
+ 
+   ``./testeth`` 
+   
+This will run all test cases automatically.
+By default `testeth` will look for the test repository cloned in cpp-ethereum submodule ``cpp-ethereum/test/jsontests`` assuming that the build folder is ``cpp-ethereum/build``
+
+If environment variable ``ETHEREUM_TEST_PATH`` is set in /etc/environment file, ``testeth`` will use path to the test repo from that variable. Example:
+
+|    ``nano /etc/environment``
+|    ``ETHEREUM_TEST_PATH="/home/user/ethereum/tests"``
+
+You could always override the test path for testeth using an option:
+
+   ``./testeth -- --testpath "/path/to/the/tests"``
+   
+Note that --testpath option argument should be an absolute path.
+For a brief help on testeth command options make sure to run 
+
+   ``./testeth --help``
+
+
+Running a specific test case
+--------------------------------------------------------------------------------
+
+To run a specific test case you could use parameter ``-t`` in the command line option:
+
+    ``./testeth -t <TEST_SUITE>/<TEST_CASE>``
+
+Or just the test suite:
+
+   ``./testeth -t <TEST_SUITE>``
+   
+To run a specific test from the test case:
+
+   ``./testeth -t <TEST_SUITE>/<TEST_CASE> -- --singletest <TEST_NAME>``
+   
+Tests has cases designed for different network rules. Such as initial frontier rules, homestead rules and other fork updates. That is to make sure that your client could sync up from the very begining to the recent top block. Block fork numbers are declared in genesis config in the file:
+
+https://github.com/ethereum/cpp-ethereum/blob/develop/libethashseal/genesis/mainNetwork.cpp
+
+If you need to debug a specific test on a specific network rules use this command:
+
+   ``./testeth -t <TEST_SUITE>/<TEST_CASE> -- --singletest <TEST_NAME> --singlenet <NET_NAME>``
+   
+Currently network names <NET_NAME> are following: Frontier, Homestead, EIP150, EIP158, Byzantine, Constantinople
+
+The main test suites are <TEST_SUITE>: StateTestsGeneral, BlockchainTests, TransitionTests, TransactionTests, VMTests
+
+<TEST_CASE> correspond to a folder name in the tests repo. And <TEST_NAME> correspond to the filename in that folder describing a specific test. 
+
+StateTestsGeneral has a single transaction being executed on a given pre state to produce a post state. 
+This transaction has arguments <data>, <value>, <gasLimit>. So a single test execute transaction with different arguments to test different conditions on the same pre state. To run a transaction from the StateTestsGeneral with the specified arguments use 
```

Comment:
For a first reader `<data>` `<value>` `<gaslimit>` options do not make sense because they don't know the tests contain arrays.  Maybe put a (TODO: explain the fact that the test files contain arrays).